indico/modules/events/registration/client/js/form/fields/AccompanyingPersonsInput.jsx

```diff
@@ -0,0 +1,288 @@
+    <FinalModalForm
+      id="accompanyingperson-form"
+      onSubmit={onSubmit}
+      onClose={onClose}
+      initialValues={value}
+      header={header}
+    >
+      <FinalInput name="firstName" label={Translate.string('First Name')} required autoFocus />
+      <FinalInput name="lastName" label={Translate.string('Last Name')} required />
+    </FinalModalForm>
+  );
+}
+
+AccompanyingPersonModal.propTypes = {
+  value: PropTypes.shape({
+    id: PropTypes.string.isRequired,
+    firstName: PropTypes.string.isRequired,
+    lastName: PropTypes.string.isRequired,
+  }),
+  header: PropTypes.string.isRequired,
+  onSubmit: PropTypes.func.isRequired,
+  onClose: PropTypes.func.isRequired,
+};
+
+AccompanyingPersonModal.defaultProps = {
+  value: {
+    id: null,
+    firstName: null,
+    lastName: null,
+  },
+};
+
+// Gather all accompanying persons field's counts.
+function countAllAccompanyingPersons(items, formState) {
+  const allAccompanyingPersonFieldNames = Object.values(items)
+    .filter(f => f.inputType === 'accompanying_persons' && f.personsCountAgainstLimit)
+    .map(apf => apf.htmlName);
+  return allAccompanyingPersonFieldNames.reduce(
+    (count, field) => count + formState.values[field]?.length || 0,
+    0
+  );
+}
+
+function calculatePlaces(availablePlaces, maxPersons, personsInCurrentField, items, formState) {
+  if (availablePlaces === null) {
+    // Field does not count towards registration limit...
+    if (!maxPersons) {
+      // ...and has no person limit.
+      return [null, Infinity];
+    } else {
+      // ...and has a person limit.
+      return [personsInCurrentField, maxPersons];
+    }
+  } else {
+    // Field counts towards registration limit...
+    const personsInAllFieldsCount = countAllAccompanyingPersons(items, formState);
+
+    if (!maxPersons || maxPersons >= availablePlaces - personsInAllFieldsCount) {
+      // ...and has no person limit, or its person limit is greater than the registration limit.
+      return [personsInAllFieldsCount, availablePlaces];
+    } else {
+      // ...and has a person limit lower than the regitration limit.
+      return [personsInCurrentField, maxPersons];
+    }
+  }
+}
+
+function AccompanyingPersonsComponent({
+  value,
+  disabled,
+  onChange,
+  price,
+  availablePlaces,
+  maxPersons,
+}) {
+  const [operation, setOperation] = useState({type: null, person: null});
+  const currency = useSelector(getCurrency);
+  const totalPrice = (value.length * price).toFixed(2);
+  const items = useSelector(getItems);
+  const formState = useFormState();
+
+  const [placesUsed, placesLimit] = calculatePlaces(
+    availablePlaces,
+    maxPersons,
+    value.length,
+    items,
+    formState
+  );
+
+  const changeReducer = action => {
+    switch (action.type) {
+      case 'ADD':
+        return [...value, {id: `new:${nanoid()}`, ...action.person}];
+      case 'EDIT':
+        return [...value.map(p => (p.id === action.person.id ? action.person : p))];
```

Comment:
```suggestion
        return value.map(p => (p.id === action.person.id ? action.person : p));
```

.map() already creates a new array. no need to clone it again